client.go

```diff
@@ -925,12 +935,40 @@ func anyError(resp *http.Response, err error) error {
 }
 
 // doRequest creates and performs an http request.
-func (c *Client) doRequest(host *URI, method, path string, headers map[string]string, reader io.Reader) (*http.Response, error) {
-	req, err := makeRequest(host, method, path, headers, reader)
+func (c *Client) doRequest(host *URI, method, path string, headers map[string]string, data []byte) (*http.Response, error) {
```

Comment:
why the change from io.Reader to []byte? oh, so it can be reused. my concern is that this would prevent streaming. but then, any io.Reader could prevent us from retrying if it isn't seekable or something.

Reply:
Yeah, this seems like the easiest approach to support retries. I don't understand why it does prevent streaming though. Can't the caller of this function read some data from the reader to a byte array?

Reply:
With the io.Reader form, the caller can pass an arbitrarily-large stream to a single request, without needing to allocate storage. This caps a request at transmitting a block of data that fits comfortably in memory, and requires users to know how much data they want to send, and handle making sequential requests for sequential blocks -- and also to know how to divide the incoming data into blocks, and handle any header-like things that have to go in front of them, etcetera.

Reply:
The biggest requests are import requests, and for roaring imports we create bitmaps which should be in memory so at least for roaring imports probably we can't support streaming (without allocating). Also sending the request  body in chunks wouldn't make a difference for the server, so reading a buffer from an `io.Reader` and posting it would work, no need to deal with headers, etc.

Reply:
My $0.02 is that I'm not worried about this right now. Column-major roaring imports aren't going to stream very well, and streaming isn't supported in Pilosa since (e.g.) unmarshalBinary takes a []byte.

I would like to tackle this sort of holistically at some point, but I think the way things are now this makes sense.

Reply:
fair enough. it's a conceptual objection but i'm not sure i have an actual use case for it.